internal/testplanet/uplink.go

```diff
@@ -0,0 +1,261 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information
+
+package testplanet
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/vivint/infectious"
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+
+	"storj.io/storj/internal/memory"
+	"storj.io/storj/pkg/eestream"
+	"storj.io/storj/pkg/identity"
+	"storj.io/storj/pkg/metainfo/kvmetainfo"
+	"storj.io/storj/pkg/overlay"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/pointerdb/pdbclient"
+	"storj.io/storj/pkg/storage/buckets"
+	ecclient "storj.io/storj/pkg/storage/ec"
+	"storj.io/storj/pkg/storage/segments"
+	"storj.io/storj/pkg/storage/streams"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/pkg/stream"
+	"storj.io/storj/pkg/transport"
+	"storj.io/storj/satellite"
+)
+
+// Uplink is a general purpose
+type Uplink struct {
+	Log              *zap.Logger
+	Info             pb.Node
+	Identity         *identity.FullIdentity
+	Transport        transport.Client
+	storageNodeCount int
+}
+
+// newUplink creates a new uplink
+func (planet *Planet) newUplink(name string) (*Uplink, error) {
+	identity, err := planet.NewIdentity()
+	if err != nil {
+		return nil, err
+	}
+
+	uplink := &Uplink{
+		Log:              planet.log.Named(name),
+		Identity:         identity,
+		storageNodeCount: len(planet.StorageNodes),
```

Comment:
Pass in the argument for storage node count rather than looking it up from planet.